### PR TITLE
clientv3: hook up KV and Txn

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony

### CODE REVIEW

File: clientv3/kv.go [hunk 1]
```diff
@@ -106,6 +118,30 @@ func (kv *kv) Delete(key string) (*DeleteResponse, error) {
...
 
+func (kv *kv) Compact(rev int64) error {
+	for {
+		r := &pb.CompactionRequest{Revision: rev}
+		_, err := kv.remote.Compact(context.TODO(), r)
+		if err == nil {
+			return nil
+		}
+
+		if isRPCError(err) {
```
Comment (heyitsanthony) [inline clientv3/kv.go]: does this belong in `switchRemote`?

Comment (xiang90) [inline clientv3/kv.go]: not really. if this is a RPC error, we should return the error directly. Or this is a network error, then we try to switchRemote etcd.

Comment (heyitsanthony) [inline clientv3/kv.go]: `isRPCError` will probably always precede `switchRemote`, so the `isRPCError` check could be folded into `switchRemote`, but not a big deal

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**